packages/smart-contracts/src/contracts/BatchConversionPayments.sol

```diff
@@ -0,0 +1,409 @@
+   *   _path Conversion path
+   *   _paymentReference References of the payment related
+   *   _feeAmount The amount in fiat of the payment fee
+   *   _maxToSpend Amounts max in token that we can spend on the behalf of the user:
+   *               it includes fee proxy but NOT the batchConversionFee
+   *   _maxRateTimespan Max times span with the oldestrate, ignored if zero
+   */
+  struct RequestInfo {
+    address _recipient;
+    uint256 _requestAmount;
+    address[] _path;
+    bytes _paymentReference;
+    uint256 _feeAmount;
+    uint256 _maxToSpend;
+    uint256 _maxRateTimespan;
+  }
+
+  /**
+   * @dev It is the structure of the input for the function from contract BatchPaymentsPublic
+   */
+  struct RequestsInfoParent {
+    address[] _tokenAddresses;
+    address[] _recipients;
+    uint256[] _amounts;
+    bytes[] _paymentReferences;
+    uint256[] _feeAmounts;
+  }
+
+  /**
+   * @dev Used by batchRouter to hold information for any kind of request.
+   *  - paymentNetworkId requests are grouped by paymentType to be paid with the appropriate function.
+   *    More details in batchRouter description.
+   *  - requestsInfo all informations required for conversion requests to be paid (=> paymentNetworkId equal 0 or 3)
+   *  - requestsInfoParent all informations required for None-conversion requests to be paid
+   *    (=> paymentNetworkId equal 1, 2, or 4)
+   */
+  struct MetaRequestsInfo {
+    uint256 paymentNetworkId;
+    RequestInfo[] requestsInfo;
+    RequestsInfoParent requestsInfoParent;
+  }
+
+  struct Path {
+    address[] _path;
+    uint256 rate;
+    uint256 decimals;
+  }
+
+  /**
+   * @param _paymentErc20FeeProxy The address to the ERC20 payment proxy to use.
+   * @param _paymentEthFeeProxy The address to the Ethereum payment proxy to use.
+   * @param _paymentErc20ConversionFeeProxy The address of the ERC20 Conversion payment proxy to use.
+   * @param _paymentEthConversionFeeProxy The address of the ETH Conversion payment proxy to use.
+   * @param _chainlinkConversionPathAddress The address of the conversion path contract
+   * @param _owner Owner of the contract.
+   */
+  constructor(
+    address _paymentErc20FeeProxy,
+    address _paymentEthFeeProxy,
+    address _paymentErc20ConversionFeeProxy,
+    address _paymentEthConversionFeeProxy,
+    address _chainlinkConversionPathAddress,
+    address _owner
+  ) BatchPaymentsPublic(_paymentErc20FeeProxy, _paymentEthFeeProxy, _owner) {
+    paymentErc20FeeProxy = IERC20FeeProxy(_paymentErc20FeeProxy);
+    paymentEthFeeProxy = IEthereumFeeProxy(_paymentEthFeeProxy);
+
+    conversionPaymentProxy = IERC20ConversionProxy(_paymentErc20ConversionFeeProxy);
+    conversionPaymentEthProxy = IEthConversionProxy(_paymentEthConversionFeeProxy);
+    chainlinkConversionPath = ChainlinkConversionPath(_chainlinkConversionPathAddress);
+    transferOwnership(_owner);
+
+    basicFee = 0;
+    batchFee = 0;
+    batchConversionFee = 0;
+  }
```

Comment:
Can you explain the difference between these 3 kinds of fees please ? 
Batch fees are not the same for ERC20 payments and ERC20WithConversion payments ?

Reply:
I did this choice to keep flexibilty.
basicFee are the one e

Reply:
I have just added: 
// Between 0 and 10000,fees applied for basic invoice, 0.1% at Request Finance
  uint256 public basicFee;

I did this choice to keep flexibilty.
basicFee are set for years at .1%.
We don't know yet if we are gonna apply the same rate for batch "basic" and batch "conversion". The second one required more expertise, which could justify a higher fee in the future (i.e: batch = 0.2% and batchConv = 0.3%).

But if we are sure to keep it this way, I can change it.
(I am cautious, because redeploy these contracts would be very painful for customers, they would have to "approve" again the proxies)